invenio_users_resources/records/api.py

```diff
@@ -276,31 +276,33 @@ def add_group(self, group_name):
         user = self.model.model_obj
         if user is None:
             return False
-        account_user = current_datastore.get_user(user.id)
-        current_roles = (
-            [role.name for role in account_user.roles] if account_user.roles else []
-        )
         user, role = current_datastore._prepare_role_modify_args(user, group_name)
-        return current_datastore.add_role_to_user(user, role)
+        result = current_datastore.add_role_to_user(user, role)
+        current_datastore.commit()
```

Comment:
I think it's best if you add `@unit_of_work()` to the service so that we ensure the transaction is rolled back if anything happens in between.